var/spack/repos/builtin/packages/py-setuptools/package.py

```diff
@@ -30,14 +30,15 @@ class PySetuptools(PythonPackage):
        upgrading, installing, and uninstalling Python packages."""
 
     homepage = "https://pypi.python.org/pypi/setuptools"
-    url      = "https://pypi.python.org/packages/source/s/setuptools/setuptools-11.3.tar.gz"
+    url      = "https://pypi.io/packages/source/s/setuptools/setuptools-25.2.0.tar.gz"
 
-    version('25.2.0', 'a0dbb65889c46214c691f6c516cf959c',
-            url="https://pypi.python.org/packages/9f/32/81c324675725d78e7f6da777483a3453611a427db0145dfb878940469692/setuptools-25.2.0.tar.gz")
+    version('34.2.0', '41b630da4ea6cfa5894d9eb3142922be',
+            url="https://pypi.io/packages/source/s/setuptools/setuptools-34.2.0.zip")
```

Comment:
I think that's the way things will have to end up going. `py-six` and `py-packaging` have that exact same try-except import statement. If the setuptools devs don't get back to me, I'll reach out to appdirs and see what they think. We can add a patch for now that replaces the import with a try-except import. Shouldn't be hard to test.

Alternatively, we can stop updating setuptools, but no one wants that 😆